packages/gatsby/src/schema/types/pagination.js

```diff
@@ -69,7 +77,15 @@ const getPagination = ({ schemaComposer, typeComposer }) => {
       resolve: group,
     },
   }
-  return createPagination({ schemaComposer, typeComposer, fields, typeName })
+  const paginationTypeComposer = createPagination({
+    schemaComposer,
+    typeComposer,
+    fields,
+    typeName,
+  })
+  paginationTypeComposer.makeFieldNonNull(`distinct`)
+  paginationTypeComposer.makeFieldNonNull(`group`)
```

Comment:
Does it apply it to Array too?

Reply:
Yes. This is what we have now:
```graphql
type FileConnection {
  totalCount: Int!
  edges: [FileEdge!]!
  pageInfo: PageInfo!
  distinct(field: FileFieldsEnum!): [String]!
  group(skip: Int, limit: Int, field: FileFieldsEnum!): [FileGroupConnection]!
}

type FileEdge {
  next: File
  node: File!
  previous: File
}

type FileGroupConnection {
  totalCount: Int!
  edges: [FileEdge!]!
  pageInfo: PageInfo!
  field: String!
  fieldValue: String
}
```

Reply:
One thing I missed though is making `internal` nonnull on Node interface

Reply:
`distinct` should be `[String!]!`, same for `[FileGroupConnection!]!`

Reply:
Hm should they? I did this on purpose because they can be empty arrays, for example with:
```graphql
{
  allFile {
    distinct(field:parent___id)
    group(field:parent___id) {
      edges {
        node {
          id
        }
      }
    }
  }
}
```

Reply:
I guess you're right though, this *should* show `null` as a `distinct` value.

Reply:
So: `distinct` should be `[String]!` (can include null), and `group` should be  `[FileGroupConnection!]!` (with fieldValue being nullable), right?

Reply:
Yes :+1:

Reply:
Hm, sorry, this still not 100% clear to me:

* `distinct` is clear: we mirror the behavior of master which filters out `null` values, so this can be an empty array -> therefore `[String]!`
* `group` is weird though, because it uses `lodash`'s `groupBy`, which apparently treats `null` and `"null"` (string) the same.
`_.groupBy([{ field: null }, { field: "null" }], 'field')`
will return
`{ null: [ { field: null }, { field: 'null' } ] }`